js/tests/unit/scrollspy.js

```diff
@@ -142,6 +142,48 @@ $(function () {
       .then(function () { return testElementIsActiveAfterScroll('#li-2', '#div-2') })
   })
 
+  QUnit.test('should add the active class correctly when there are nested elements at 0 scroll offset', function (assert) {
+    var navbarHtml =
+        '<nav class="navbar">'
+      + '<ul class="nav">'
+      + '<li id="li-1"><a href="#div-1">div 1</a>'
+      + '<ul>'
+      + '<li id="li-2"><a href="#div-2">div 2</a></li>'
+      + '</ul>'
+      + '</li>'
+      + '</ul>'
+      + '</nav>'
+    var contentHtml =
+        '<div class="content" style="position: absolute; top: 0px; overflow: auto; height: 50px">'
+      + '<div id="div-1" style="padding: 0; margin: 0">'
+      + '<div id="div-2" style="height: 200px; padding: 0; margin: 0">div 2</div>'
+      + '</div>'
+      + '</div>'
+
+    $(navbarHtml).appendTo('#qunit-fixture')
+    var $content = $(contentHtml)
+      .appendTo('#qunit-fixture')
+      .bootstrapScrollspy({ offset: 0, target: '.navbar' })
+
+    var testElementsAreActiveAfterScroll = function (element) {
+      var deferred = $.Deferred()
+      var scrollHeight = $content.scrollTop() + 10
+      var done = assert.async()
+      $content.one('scroll', function () {
+        assert.ok($('#li-1').hasClass('active'), 'element' + element)
```

Comment:
Please use more descriptive assertion messages, e.g. 'nav item for outer element has "active" class'